draft-ietf-tls-tls13.md

```diff
@@ -2950,6 +2956,120 @@ complete picture is that ECDH is employed with a non-trivial KDF
 because TLS does not directly use the premaster secret for anything
 other than for computing the master secret.)
 
+## Update Protocol
+
+TLS supports an Update protocol which allows endpoints to update the
+cryptographic state of the connection after the handshake has been
+completed. The Update protocol can currently be used for the following tasks:
+
+- Refreshing cryptographic keying material.
+
+- Establishing a shared key which can be used to resume future
+  sessions.
+
+In versions of TLS prior to TLS 1.3, these functions were part of the handshake
+and/or required renegotiation. In TLS 1.3 they have been moved into a separate
+subprotocol. Future versions of this specification may add new functions to
+the Update protocol.
+
+
+### General Update Protocol Behavior
+
+In general, Update messages have the following structure:
+
+       enum { rekey(1), ack(2),  authentication(3), (255) } UpdateType;
+
+       struct {
+           UpdateType msg_type;       /* update type */
+           uint24 length;             /* bytes in message */
+
+           switch (msg_type) {
+             case ack:
+               UpdateAck ack;
+ 
+             case rekey:
+               UpdateRekey rekey;
+  
+             /* This structure may be extended */
+           }
+        } Update;
+
+Whenever a site sends an Update message, it updates its own master secret
+as follows:
+
+      X_master_secret_{i+1} = PRF(X_master_secret_i, "master_secret_update",
+                                  Update_message)
```

Comment:
This could use "client master secret" or "server master secret" as above.  You should also probably be more consistent in the naming of these things.

I've argued before for a table somewhere listing both the PRF and the signature discriminator strings.  Might be valuable as an appendix.  Once this settles, perhaps I can generate you a PR.